tools/src/tokenize.rs

```diff
@@ -266,7 +266,77 @@ pub fn tokenize_c_like(string: &String, spec: &LanguageSpec) -> Vec<Token> {
                 tokens.push(Token {start: start, end: peek_pos(), kind: TokenKind::StringLiteral});
             }
             next_token_maybe_regexp_literal = true;
-        } else if ch == '\'' || ch == '"' {
+        } else if ch == 'R' {
+            // Handle raw literals.
+            assert!(peek_char() == '"');
+            get_char();
+
+            // Read delimiter.
+            loop {
+                let c = peek_char();
+                if c == '(' {
+                    break;
+                }
+
+                if c == ')' || c == '\\' || is_whitespace(c) {
+                    writeln!(&mut std::io::stderr(), "Illegal character in raw string delimiter").unwrap();
+                    return tokens;
+                }
+
+                get_char();
```

Comment:
Instead of using peek_char followed by get_char, you might try saving both results of get_char. I think it might simplify the code in some cases. For example, I think it's fine to consume the '(' in this loop. You could save its position(the first result of get_char)  in a variable and then use that in the slice that you take below.